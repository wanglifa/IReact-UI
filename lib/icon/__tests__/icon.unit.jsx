import React from 'react'
import renderer from 'react-test-renderer'
import Icon from '../icon'
import { mount } from 'enzyme'

describe('Icon', () => {
  it('render successfully', () => {
    //渲染一个Button，因为Button是一个对象所以我们可以把它转成json
    const json = renderer.create(<Icon name="alipay"/>).toJSON()
    //期待它去匹配Snapshot
    expect(json).toMatchSnapshot()
  })
  it('onClick', () => {
    const fn = jest.fn()
    const component = mount(<Icon name="alipay" onClick={fn}/>)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled
  })
})