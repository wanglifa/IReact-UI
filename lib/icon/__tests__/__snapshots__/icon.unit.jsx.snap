// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon render successfully 1`] = `
<svg
  className="ireact-icon qq"
>
  <use
    xlinkHref="#alipay"
  />
</svg>
`;

exports[`Icon 是个svg 1`] = `
<svg
  className="ireact-icon qq"
>
  <use
    xlinkHref="#alipay"
  />
</svg>
`;
